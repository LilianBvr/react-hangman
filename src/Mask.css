.mask{
  margin:0;
  padding:2vh;

  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: center;
  
  border-radius: 20px;
  background-color: #EEE; 
}

.mask > .letter{
  margin: 0 2vw 0 2vw;
  font-size: 2em;
  font-family:courier;

  color:#888;
}

.mask > .letter.hidden{

}