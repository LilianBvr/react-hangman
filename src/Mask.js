import React from 'react'
import PropTypes from 'prop-types'

import './Mask.css'

const Mask = ({ mask }) => (
	
	<div className='mask' >
		{
			mask.map(( letter, index ) => (
				<span 
					className={`letter ${ letter==='_' ? 'hidden' : '' }`} 
					key={index}
				>
					{letter}
				</span>
			))
		}	
	</div>
)

Mask.propTypes = {
	mask: PropTypes.array
}

export default Mask