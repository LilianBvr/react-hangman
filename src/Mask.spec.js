import React from 'react'
import { mount, shallow } from 'enzyme'
import { expect } from 'chai'

import Mask from './Mask'

describe('<Mask />', () => {

    it('should render and match its reference snapshot', () => {
        expect(shallow(<Mask mask={Array.from('T__T')} />)).to.matchSnapshot()
    })
})