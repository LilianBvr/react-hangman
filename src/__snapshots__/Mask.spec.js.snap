// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mask /> should render and match its reference snapshot 1`] = `
<div
  className="mask"
>
  <span
    className="letter "
    key="0"
  >
    T
  </span>
  <span
    className="letter hidden"
    key="1"
  >
    _
  </span>
  <span
    className="letter hidden"
    key="2"
  >
    _
  </span>
  <span
    className="letter "
    key="3"
  >
    T
  </span>
</div>
`;
